var/spack/repos/builtin/packages/py-meep/package.py

```diff
@@ -0,0 +1,75 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyMeep(Package):
+    """Python-meep is a wrapper around libmeep. It allows the scripting of
+    Meep-simulations with Python"""
+
+    homepage = "https://launchpad.net/python-meep"
+    url      = "https://launchpad.net/python-meep/1.4/1.4/+download/python-meep-1.4.2.tar"
+
+    version('1.4.2', 'f8913542d18b0dda92ebc64f0a10ce56')
+
+    variant('mpi', default=True, description='Enable MPI support')
+
+    extends('python')
+    depends_on('py-numpy', type=nolink)
+    depends_on('py-scipy', type=nolink)
+    depends_on('py-matplotlib', type=nolink)
+
+    depends_on('mpi', when='+mpi')
+    depends_on('meep~mpi', when='~mpi')
+    depends_on('meep+mpi', when='+mpi')
+
+    # As of SWIG 3.0.3, Python-style comments are now treated as
+    # pre-processor directives. Use older SWIG. But not too old,
+    # or else it can't handle newer C++ compilers and flags.
+    depends_on('swig@1.3.39:3.0.2')
+
+    def install(self, spec, prefix):
+        setup = 'setup-mpi.py' if '+mpi' in spec else 'setup.py'
+
+        include_dirs = [
+            spec['meep'].prefix.include,
+            spec['py-numpy'].include
+        ]
+
+        library_dirs = [
+            spec['meep'].prefix.lib
+        ]
+
+        if '+mpi' in spec:
+            include_dirs.append(spec['mpi'].prefix.include)
+            library_dirs.append(spec['mpi'].prefix.lib)
```

Comment:
I don't think this actually works. The package is linked to MPI properly:
```
$ ldd py-meep-1.4.2-ikjdpnikleqeaxvncnag37bhhdnfafoj/lib/python2.7/site-packages/_meep_mpi.so 
...
	libmpi.so.20 => /blues/gpfs/home/software/spack-0.9.1/opt/spack/linux-centos6-x86_64/gcc-6.1.0/openmpi-2.0.0-tapt7aozvnpq4m7tf7cbsa5zefy4kc6s/lib/libmpi.so.20 (0x00002b0cbd86b000)
...
```
but when I import it:
```
$ python
Python 2.7.12 (default, Jul 21 2016, 15:23:03) 
[GCC 6.1.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import meep_mpi
Neither libmpi.so.0 nor libmpi.so found. Fatal error.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/soft/spack-0.9.1/opt/spack/linux-centos6-x86_64/gcc-6.1.0/python-2.7.12-hfkksvc2jgtlvkjagr33k6khrnj7yfqt/lib/python2.7/site-packages/meep_mpi.py", line 5088, in <module>
    raise e
OSError: libmpi.so: cannot open shared object file: No such file or directory
```
it can't find MPI.